api/desecapi/tests/test_rrsets.py

```diff
@@ -239,14 +248,155 @@ def test_create_my_rr_sets_empty_payload(self):
         response = self.client.post_rr_set(self.my_empty_domain.name)
         self.assertContains(response, 'No data provided', status_code=status.HTTP_400_BAD_REQUEST)
 
-    def test_create_my_rr_sets_unknown_type(self):
-        for _type in ['AA', 'ASDF']:
-            with self.assertPdnsRequests(
-                    self.request_pdns_zone_update_unknown_type(name=self.my_domain.name, unknown_types=_type)
-            ):
-                response = self.client.post_rr_set(self.my_domain.name, records=['1234'], ttl=3660, type=_type)
+    def test_create_my_rr_sets_canonical_content(self):
+        # TODO fill in more examples
+        datas = [
```

Comment:
should be a dict

Reply:
no, dicts do not allow duplicate keys. (Assuming you meant that the record type is the key.)

Reply:
Well, you don't have duplicates, so I didn't expect that that should be supported :)

Fine with me. -- Anyways, [this `defaultdict` idea](https://stackoverflow.com/a/6813689/6867099) may be instructive (not implying that it should be used here).

Reply:
I prefer multiple lines with the same record type over one line where either the elements in the list have alternating meaning (input, canonical expectation, ...) or is a nested list, like suggested in `defaultdict`.